utils/node.go

```diff
@@ -2,22 +2,98 @@ package utils
 
 import (
 	"fmt"
+	"strings"
+	"time"
 
 	coretypes "github.com/cometbft/cometbft/rpc/core/types"
 	"github.com/forbole/juno/v6/node"
+	"github.com/rs/zerolog/log"
 )
 
+// isRetryableError checks if an error is retryable (timeout, gateway error, etc.)
+func isRetryableError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	errMsg := err.Error()
+	retryableErrors := []string{
+		"504 Gateway Time-out",
+		"503 Service Unavailable",
+		"502 Bad Gateway",
+		"timeout",
+		"connection refused",
+		"connection reset",
+		"EOF",
+		"invalid character '<'", // HTML error pages
+	}
+
+	for _, retryable := range retryableErrors {
+		if strings.Contains(errMsg, retryable) {
+			return true
+		}
+	}
+
+	return false
+}
+
 // QueryTxs queries all the transactions from the given node corresponding to the given query
+// with retry logic for transient errors like timeouts
 func QueryTxs(node node.Node, query string) ([]*coretypes.ResultTx, error) {
 	var txs []*coretypes.ResultTx
 
 	page := 1
 	perPage := 100
 	stop := false
+
+	const maxRetries = 10
```

Comment:
in coreum-tools we had retrial logic implemented. 
Could you pls check if we can use it ?